includes/class-wc-admin-reports-sync.php

```diff
@@ -211,47 +183,86 @@ public static function orders_lookup_update_init() {
 
 	/**
 	 * Init order/product lookup tables update (in batches).
+	 *
+	 * @param integer|boolean $days Number of days to process.
+	 * @param boolean         $skip_existing Skip exisiting records.
 	 */
-	public static function orders_lookup_batch_init() {
-		$batch_size  = self::get_batch_size( self::ORDERS_BATCH_ACTION );
-		$order_query = new WC_Order_Query(
-			array(
-				'return'   => 'ids',
-				'limit'    => 1,
-				'paginate' => true,
-			)
-		);
-		$result      = $order_query->get_orders();
+	public static function orders_lookup_batch_init( $days, $skip_existing ) {
+		$batch_size = self::get_batch_size( self::ORDERS_BATCH_ACTION );
+		$orders     = self::get_orders( 1, 1, $days, $skip_existing );
 
-		if ( 0 === $result->total ) {
+		if ( 0 === $orders->total ) {
 			return;
 		}
 
-		$num_batches = ceil( $result->total / $batch_size );
+		$num_batches = ceil( $orders->total / $batch_size );
 
-		self::queue_batches( 1, $num_batches, self::ORDERS_BATCH_ACTION );
+		self::queue_batches( 1, $num_batches, self::ORDERS_BATCH_ACTION, array( $days, $skip_existing ) );
 	}
 
 	/**
-	 * Process a batch of orders to update (stats and products).
+	 * Get the order IDs and total count that need to be synced.
 	 *
-	 * @param int $batch_number Batch number to process (essentially a query page number).
-	 * @return void
+	 * @param int      $limit Number of records to retrieve.
+	 * @param int      $page  Page number.
+	 * @param int|bool $days Number of days prior to current date to limit search results.
+	 * @param bool     $skip_existing Skip already imported orders.
 	 */
-	public static function orders_lookup_process_batch( $batch_number ) {
-		$batch_size  = self::get_batch_size( self::ORDERS_BATCH_ACTION );
-		$order_query = new WC_Order_Query(
-			array(
-				'return'  => 'ids',
-				'limit'   => $batch_size,
-				'page'    => $batch_number,
-				'orderby' => 'ID',
-				'order'   => 'ASC',
+	public static function get_orders( $limit = 10, $page = 1, $days = false, $skip_existing = false ) {
+		global $wpdb;
+		$where_clause = '';
+		$offset       = $page > 1 ? $page * $limit : 0;
+
+		if ( $days ) {
+			$days_ago      = date( 'Y-m-d H:i:s', time() - ( DAY_IN_SECONDS * $days ) );
```

Comment:
I find this a bit confusing. If I activate the _Last 7 days_ import on April 16 11am, I would expect all orders from April 9 to be included, even if they happened at 10am. With this implementation, it would only select the orders occurred in the last 168 hours (7 * 24), which would not include an order from April 9 10am, right?

I don't mind leaving it as-is if I'm the only one with those expectations, but just wanted to write it down in case any of you agree. :slightly_smiling_face:

Reply:
I would expect the number of hours personally, but I recognize that may be too much of a developer mindset.

I think your expectations are probably more in line with what most people would think.  Will add this in.

Reply:
Updated in https://github.com/woocommerce/woocommerce-admin/pull/2034/commits/36a12000f84e6dbdfc16080309fccf7abb23a7a0